tests/test_not.py

```diff
@@ -0,0 +1,154 @@
+import pytest
+
+from boolean import BooleanAlgebra
+
+class TestNOT:
+
+    def test_raises(self):
+        algebra = BooleanAlgebra()
+
+        with pytest.raises(TypeError):
+            algebra.NOT()
+
+        with pytest.raises(TypeError):
+            algebra.NOT('a', 'b')
+
+    def test_true_and_false(self):
+        algebra = BooleanAlgebra()
+
+        assert algebra.TRUE is (algebra.NOT(algebra.FALSE)).simplify()
+        assert algebra.FALSE is (algebra.NOT(algebra.TRUE)).simplify()
+
+        assert algebra.TRUE == (algebra.NOT(algebra.FALSE)).simplify()
+        assert algebra.FALSE == (algebra.NOT(algebra.TRUE)).simplify()
+
+    def test_isliteral(self):
+        algebra = BooleanAlgebra()
+
+        s = algebra.Symbol(1)
+
+        # negation of a literal is still a literal
+        assert algebra.NOT(s).isliteral
+        # negation of a non-literal is still a non-literal
+        assert not algebra.parse('~(a|b)').isliteral
+
+    def test_literals_0(self):
+        algebra = BooleanAlgebra()
+
+        a = algebra.Symbol('a')
+        b = ~a
+
+        assert a.isliteral
+        assert b.isliteral
+
+        assert a in a.literals
+        assert b in b.literals
+
+        assert len(a.literals) == 1
+        assert len(b.literals) == 1
+
+    def test_literals_1(self):
+        algebra = BooleanAlgebra()
+
+        expression = algebra.parse('~(a&a)')
+
+        assert not expression.isliteral
+
+        assert algebra.Symbol('a') in expression.literals
+        assert len(expression.literals) == 1
+
+    def test_literals_2(self):
+        algebra = BooleanAlgebra()
+
+        expression = algebra.parse('~(a&a)', simplify=True)
+
+        assert expression.isliteral
+        assert expression == algebra.NOT(algebra.Symbol('a'))
+
+    def test_literalize(self):
+        parse = BooleanAlgebra().parse
+
+        assert parse('~a').literalize() == parse('~a')
+        assert parse('~(a&b)').literalize() ==  parse('~a|~b')
+        assert parse('~(a|b)').literalize() ==  parse('~a&~b')
+
+    def test_invert_eq_not(self):
+        algebra = BooleanAlgebra()
+
+        a = algebra.Symbol('a')
+
+        assert ~a == ~a
+        assert ~a == algebra.NOT(a)
+
+    def test_simplify(self):
+        algebra = BooleanAlgebra()
+
+        a = algebra.Symbol('a')
+
+        assert a == a.simplify()
+        assert a == (~~a).simplify()
+        assert a == (~~ ~~a).simplify()
```

Comment:
Actually, no, that space was intentional. It was there before:

https://github.com/bastikr/boolean.py/pull/70/commits/bfb0b31dbb2ced65edf6f64d42c7171c2daa2475#diff-b03f7319ce9e9d763799954c9b76efa9L241